core/domain/learner_progress_services.py

```diff
@@ -751,6 +1124,87 @@ def _get_filtered_incomplete_exp_summaries(
     return filtered_incomplete_exp_summaries, nonexistent_incomplete_exp_ids
 
 
+def get_all_incomplete_story_ids(user_id):
+    """Returns a list with the ids of all the stories partially completed
+    by the user.
+
+    Args:
+        user_id: str. The id of the learner.
+
+    Returns:
+        list(str). A list of the ids of the stories partially completed by
+        the learner.
+    """
+    incomplete_activities_model = (
+        user_models.IncompleteActivitiesModel.get(user_id, strict=False))
+
+    if incomplete_activities_model:
+        incomplete_activities = _get_incomplete_activities_from_model(
+            incomplete_activities_model)
+
+        return incomplete_activities.story_ids
+    else:
+        return []
```

Comment:
You don't really need this else, since the if has a return statement, right? You can remove it.

Reply:
Done.

Reply:
You need the `return []`. I meant, you don't need the `else` statement, right? It can just be `return []`.

Reply:
Marking this comment as well about removing else.

Reply:
Done.